webapp/channels/src/components/admin_console/server_logs/logs.tsx

```diff
@@ -121,13 +125,23 @@ export default class Logs extends React.PureComponent<Props, State> {
 
     performSearch = debounce(() => {
         const {search} = this.state;
+
+        // Excluding job_id, level, and timestamp from search
+        const excludedKeys = new Set(['job_id', 'level', 'timestamp']);
+
         const filteredLogs = this.props.logs.filter((log) => {
-            // to be improved
-            return `${log.caller}${log.msg}${log.worker}${log.worker}`.toLowerCase().includes(search.toLowerCase());
+            const logEntriesForSearch = Object.entries(log).filter(([logKey]) => !excludedKeys.has(logKey));
+            return logEntriesForSearch.some(([, logValue]) => {
+                return String(logValue).toLowerCase().includes(search.toLowerCase());
+            });
         });
         this.setState({filteredLogs});
     }, 200);
 
+    componentWillUnmount(): void {
+        this.performSearch.cancel();
```

Comment:
I had to add this debounce cancel to avoid this error when executing tests: `Unexpected console logsWarning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in %s.%s,the componentWillUnmount method`